commit 21fc11a15051fcf934b8b25fe4a287442d253288
Author: Gawain Lynch <gawain.lynch@gmail.com>
Date:   Sat Apr 4 13:32:58 2015 +0100

    [OAuth2] Cast grant to string

diff --git a/lib/League/OAuth2/Provider/AbstractProvider.php b/lib/League/OAuth2/Provider/AbstractProvider.php
index 38e1edc..b194950 100644
--- a/lib/League/OAuth2/Provider/AbstractProvider.php
+++ b/lib/League/OAuth2/Provider/AbstractProvider.php
@@ -170,7 +170,7 @@ abstract class AbstractProvider implements ProviderInterface
             'client_id'     => $this->clientId,
             'client_secret' => $this->clientSecret,
             'redirect_uri'  => $this->redirectUri,
-            'grant_type'    => $grant,
+            'grant_type'    => (string) $grant,
         ];
 
         $requestParams = $grant->prepRequestParams($defaultParams, $params);
